src/plugins/hidsim/gui/vmi_win_gui_parser.cpp

```diff
@@ -327,10 +327,10 @@ bool filter_wnd_text(wchar_t* text)
                 continue;
 
             /* Calculates maximum possible displacement */
-            max_disp = tlen - (MAX_DISPLACEMENT * llen) < 0 ?
-                tlen - llen + 1 : MAX_DISPLACEMENT;
+            max_disp = tlen - llen < MAX_DISPLACEMENT ?
+                tlen - llen : MAX_DISPLACEMENT;
 
```

Comment:
That's not an issue: I can not fully understand the logic of this replacement. Is this code equivalent to previous version?

Reply:
> Is this code equivalent to previous version?

It's not equivalent, it actually corrects the calculation of the amount of possible shifts.   

In the housing function it is checked, whether the target-string (defined in `vmi_win_gui_filter_definitions.h`) can be found within the first  `MAX_DISPLACEMENT` positions of the candidate window's text. 

This statement here
```
max_disp = tlen - llen < MAX_DISPLACEMENT ?
                tlen - llen : MAX_DISPLACEMENT;
``` 
calculates the maximum displacement to take into account -- meaning how many characters to shift through, which is dependent of the overall length of the window's text and the length of the target text. 
E.g.: Looking at the window-text of `&Agree` and the target-text `agree`,  then two comparisons will be conducted (comparing "&Agre" with "agree" and "Agree" with "agree").  

Hope, this clarifies the operations.